### PR TITLE
Migrate PluginUnderTestMetadata task to provider API

### PR DESCRIPTION
**Pull request by ov7a**
### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/PluginUnderTestMetadata.java [hunk 1]
```diff
@@ -45,24 +45,21 @@ public abstract class PluginUnderTestMetadata extends DefaultTask {
...
     public static final String METADATA_FILE_NAME = "plugin-under-test-metadata.properties";
-    private final ConfigurableFileCollection pluginClasspath = getProject().files();
-    private final DirectoryProperty outputDirectory = getProject().getObjects().directoryProperty();
 
     /**
      * The code under test. Defaults to {@code sourceSets.main.runtimeClasspath}.
+     *
+     * Note: this is not a @Classpath since the ablosute paths are written to the metadata file.
      */
-    @Classpath
```
Comment (ov7a) [inline platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/PluginUnderTestMetadata.java:54]: ❓ Is it ok to change `@Classpath` to `@InputFiles` here?

Comment (tresat) [reply platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/PluginUnderTestMetadata.java:54]: This seems correct to me.

File: platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/PluginUnderTestMetadata.java [hunk 2]
```diff
@@ -78,7 +75,10 @@ public void generate() {
 
     private String implementationClasspath() {
         StringBuilder implementationClasspath = new StringBuilder();
-        Joiner.on(File.pathSeparator).appendTo(implementationClasspath, getPaths());
+        Joiner.on(File.pathSeparator).appendTo(
+            implementationClasspath,
+            collect(getPluginClasspath(), file -> file.getAbsolutePath().replaceAll("\\\\", "/"))
```
Comment (tresat) [inline platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/PluginUnderTestMetadata.java:80]: 🤔 `FilenameUtils.separatorsToUnix` perhaps?

**Review by tresat — APPROVED** [review]

**Comment by ov7a** [comment]
@bot-gradle test this

**Review by asodja — APPROVED** [review]
LGTM

**Comment by ov7a** [comment]
@bot-gradle test this

**Comment by ov7a** [comment]
@bot-gradle test this

**Pull request merged**